File: myoung/programmers/64065/programmers_64065.py
def solution(s):
    answer = []
    s = s[2:-2].split('},{')
    s.sort(key=lambda x: len(x))
    prev = []
    for i in range(len(s)):
        now = list(map(int, s[i].split(',')))
        val = [x for x in now if x not in prev]
        prev = now
        answer.append(val[0])
    return answer

print(solution("{{2},{2,1},{2,1,3},{2,1,3,4}}"))

File: HanJaehee/programmers/64065/programmers64065.java
import java.util.*;
class Solution {
    
    public int[] solution(String s) {
        s = s.substring(2, s.length()-2);
        String[] sList = s.split("\\},\\{");
        Arrays.sort(sList, (a,b)-> a.length() - b.length());
        Set<Integer> set = new HashSet<>();
        
        int[] result = new int[sList.length];
        int idx = 0;
        for(String tmpString : sList){
            String[] tmpList = tmpString.split(",");
            for(String tmp : tmpList){
                int num = Integer.parseInt(tmp);
                if(!set.contains(num)){
                    result[idx++] = num;
                    set.add(num);
                }
            }
        }
        
        return result;
    }
}